app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -2,10 +2,34 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   include PreferencesController
 
   require_authentication :all, scopes: [:project]
+  extra_schema_actions :update_settings
   schema_type :json_schema
+  before_action :require_project_ownership, only: [:update_project_settings]
+
+
+  def update_settings
+    upp = UserProjectPreference.find_or_initialize_by(project_id: params_for[:project_id],
+                                                      user_id: params_for[:user_id])
+    upp.settings.merge! params_for[:settings]
+    if upp.save!
+      render status: :ok, nothing: true
+    else
+      render status: :unprocessable_entity, nothing: true
+    end
+  end
 
   private
 
+  def project
+    Project.find params_for[:project_id]
+  end
+
+  def require_project_ownership
+    unless project.owner?(api_user.user)
+      render json: { error: 'not authorized' }, status: :not_authorized
```

Comment:
Your custom error can contain a message as per normal and the controller should put it into the response body correctly [here](https://github.com/zooniverse/Panoptes/blob/18a44e0c7e6775c4a9ae1b794ce87ed86e485492/app/controllers/concerns/json_api_render.rb#L24)